src/components/ec-option-card/ec-option-card.vue

```diff
@@ -0,0 +1,196 @@
+    @click="setClickAction()"
+  >
+    <span
+      data-test="ec-option-card__title-img"
+    >
+      <ec-icon
+        v-if="iconName && title"
+        data-test="ec-option-card__icon"
+        class="ec-option-card__icon"
+        :class="getOptionIconClass()"
+        :name="iconName"
+        :alt="title"
+        :size="24"
+      /><p class="ec-option-card__title">{{ title }}</p>
+    </span>
+    <p
+      v-if="caption"
+      class="ec-option-card__caption"
+      :class="getcaptionClass()"
+      data-test="ec-option-card__caption"
+    >{{ caption }}</p>
+  </component>
+</template>
+
+<script setup lang="ts">
+import {
+  computed,
+  toRefs,
+  useAttrs,
+} from 'vue';
+
+import EcIcon from '../ec-icon';
+import type { IconName } from '../ec-icon/types';
+import { OptionCardEvent, OptionCardType } from './types';
+
+const attrs = useAttrs();
+
+interface OptionCardProps {
+  isDisabled?: boolean,
+  title: string,
+  caption?: string,
+  iconName?: IconName,
+  type?: OptionCardType,
+  to?: string,
+  href?: string,
+}
+
+const props = defineProps<OptionCardProps>();
+const emit = defineEmits<{(e: 'click'): void}>();
+
+const {
+  isDisabled,
+  title,
+  caption,
+  iconName,
+  type,
+  to,
+  href,
+} = toRefs(props);
+
+const componentTag = computed(() => {
+  if (to?.value) {
+    return 'router-link';
+  } if (href?.value) {
+    return 'a';
+  }
+  return 'button';
+});
+
+const isOptionCardDisabled = computed(() => ({ disabled: isDisabled?.value }));
+
+const optionCardNavigationType = computed(() => (to?.value ? { to: to.value } : { href: href?.value }));
+
+const componentProps = computed(() => ({
+  ...isOptionCardDisabled.value,
+  ...(componentTag.value === 'button' && { type: componentTag.value }),
+  ...optionCardNavigationType.value,
+}));
+
+function getOptionCardClasses() {
+  return {
+    'ec-option-card--disabled': Boolean(isDisabled?.value),
+    'ec-option-card--accent': type?.value === OptionCardType.OPTION_CARD_ACCENT,
+    'ec-option-card--danger': type?.value === OptionCardType.OPTION_CARD_DANGER,
+  };
+}
+
+function getOptionIconClass() {
+  return {
+    'ec-option-card__icon--disabled': Boolean(isDisabled?.value),
+    'ec-option-card__icon--accent': type?.value === OptionCardType.OPTION_CARD_ACCENT,
+    'ec-option-card__icon--danger': !isDisabled?.value && (type?.value === OptionCardType.OPTION_CARD_DANGER),
+  };
+}
+
+function getcaptionClass() {
+  return {
+    'ec-option-card__caption--disabled': Boolean(isDisabled?.value),
+    'ec-option-card__caption--accent': type?.value === OptionCardType.OPTION_CARD_ACCENT,
+    'ec-option-card__caption--danger': !isDisabled?.value && (type?.value === OptionCardType.OPTION_CARD_DANGER),
+  };
+}
+
+function setClickAction() {
+  if (componentTag.value === 'button' && !isDisabled?.value) {
+    return emit(OptionCardEvent.OPTION_CARD_CLICK);
+  }
+  return null;
+}
+</script>
+
+<style>
+:root {
+  --ec-option-card-width: 304px;
```

Comment:
card should be flexible and have no fixed size, please have a look at `ec-summary-info` for example, it should use space provided by upper level html element.

Reply:
Hey Anton, sorry can you explain this a bit more?
Why should be flexible if in the design has a fixed width? Speaking with Alba she confirmed the width will always be 304px. 
The only thing will need (in the feature) is to make it mobile friendly. Depending on the page container of course, usually a small screen is 320px so this should be fine.
Are you saying is better practice to have the parent set the width every-time? 
What's the benefit of that if we know that it must be 304px?  
Thank you!

Reply:
Hey, yes I'm saying that parent should set the width and element use all space available or needed for it. This is the best practice to have components adjustable by place of using if it's needed. And having fixed size of element is error prone approach, as we could have cases when it could break the layout.